### ISSUE TITLE
[Grid] Battery Ohmic Heat Generation + Thermal Output

### ISSUE THREAD

**Issue by anchapin** [anchapin/fluxion#2036]
## Problem

Implement electrochemical thermal model: Ohmic heat generation during charge/discharge.

## Context

Battery internal resistance converts electrical energy to heat: `Q_gen = I² * R_internal`. This heat must be accounted for in zone energy balances.

## Solution

```rust
impl BatteryStorageNode {
    pub fn ohmic_heat_generation(&self, current_amps: f64) -> HeatFlowRate {
        // Q_gen = I² * R_internal
        let i_sq = current_amps * current_amps;
        HeatFlowRate::from_watts(i_sq * self.r_internal_ohm)
    }
    
    pub fn thermal_output(&self, dt: Duration, current_amps: f64) -> HeatFlowRate {
        // Positive = heating, Negative = cooling (rare for discharge)
        let q_gen = self.ohmic_heat_generation(current_amps);
        let efficiency_loss = (current_amps.powi(2) * self.r_internal_ohm) * dt.as_secs_f64();
        q_gen
    }
}
```

## Acceptance Criteria

- [ ] Ohmic heat: `Q_gen = I² * R_internal` (Watts)
- [ ] Returns `HeatFlowRate` for direct integration with `fluxion-core`
- [ ] Unit test: 100A discharge through 0.01Ω → Q_gen = 100W
- [ ] Cell temperature rise estimate: `ΔT = Q_gen / (m * c_p)`

## Dependencies

- Issue #2016 (BatteryStorageNode core)

## Labels

fluxion-grid, battery, physics, phase-2

## References

- PLAN_NEXTGEN_FLUXION.md
